src/toast/tests/spt3g.py

```diff
@@ -0,0 +1,406 @@
+            n_set = ob.dist.samp_sets[ob.comm_rank].n_elem
+            for sset in range(first_set, first_set + n_set):
+                for chunk in ob.dist.sample_sets[sset]:
+                    n_frames += 1
+            if ob.comm_row is not None:
+                n_frames = ob.comm_row.allreduce(n_frames, op=MPI.SUM)
+            ob_n_frames[ob.name] = n_frames
+
+        return data, ob_n_frames
+
+    def test_import_export(self):
+        rank = 0
+        if self.comm is not None:
+            rank = self.comm.rank
+
+        data, ob_n_frames = self.create_data()
+
+        # The default export / import classes
+        meta_exporter = export_obs_meta(
+            noise_models=[
+                ("noise_model", "noise_model"),
+                ("el_weighted", "el_weighted"),
+            ]
+        )
+        data_exporter = export_obs_data(
+            shared_names=self.export_shared_names,
+            det_names=self.export_det_names,
+            interval_names=self.export_interval_names,
+            compress=False,
+        )
+
+        exporter = export_obs(
+            meta_export=meta_exporter,
+            data_export=data_exporter,
+        )
+
+        meta_importer = import_obs_meta(
+            noise_models=[
+                ("noise_model", "noise_model"),
+                ("el_weighted", "el_weighted"),
+            ]
+        )
+        data_importer = import_obs_data(
+            shared_names=self.import_shared_names,
+            det_names=self.import_det_names,
+            interval_names=self.import_interval_names,
+        )
+        importer = import_obs(
+            comm=data.comm.comm_group,
+            meta_import=meta_importer,
+            data_import=data_importer,
+        )
+
+        # Export the data, and make a copy for later comparison.
+        original = list()
+        g3data = list()
+        for ob in data.obs:
+            original.append(ob.duplicate(times="times"))
+            obframes = exporter(ob)
+            # There should be the original number of frame intervals plus
+            # one observation frame and one calibration frame
+            checktot = len(obframes) - 2
+            if ob.comm_row is not None:
+                checktot = ob.comm_row.allreduce(checktot, op=MPI.SUM)
+            if checktot != ob_n_frames[ob.name]:
+                msg = f"proc {data.comm.world_rank} on ob {ob.name} "
+                msg += f"has {checktot} scan frames, "
+                msg += f"not {ob_n_frames[ob.name]}"
+                print(msg, flush=True)
+                self.assertTrue(False)
+            g3data.append(obframes)
+
+        # Import the data
+        check_data = Data(comm=data.comm)
+
+        try:
+            for obframes in g3data:
+                check_data.obs.append(importer(obframes))
+
+            for ob in check_data.obs:
+                ob.redistribute(ob.comm_size, times="times")
+        except Exception as e:
```

Comment:
Should the try-except remain here after debugging?

Reply:
Removed.